### ISSUE TITLE
Request: ability to bring up and read tooltips without a mouse

### ISSUE THREAD

**Issue by TechHorseG** [nvaccess/nvda#10320]
Feature request: An option to force NVDA to read the tooltips associated with an object. If you hover over an object and a tooltip appears, NVDA can read it out. But I am talking about a method that does not use the mouse and requires no hovering. So I could just arrow down the list of files in Explorer and then press a key combo to cause NVDA to bring up and / or read the tooltip info.

As another example, in FreeFileSync if I tab through some of the buttons it'll just state "button". Whereas if I hover over that button a tooltip explaining what it does appears. But with less than perfect eyesight aiming a pointer at small buttons can get fiddly.

This would also be useful for people who do not use a mouse, as they could still read the tooltip info with a press of some keys.

**Comment by josephsl** [comment]
Hi, so this goes a step further than letting NVDA announce tool tips automatically, correct? Note that not all situations can support tool tips or mouse over text (for File Explorer, it requires reading file/folder metadata, which not all file types support). CC @MichaelDCurran

**Comment by TechHorseG** [comment]
Yes I appreciate that this might be tricky for some tooltips, if they call other data. But hopefully possible for static text tooltips?

With FFS as mentioned if I keyboard nav to a button I will just hear "button". Whereas if I mouse hover the tooltip "swap sides" will be read out.

Surely that is just a simple piece of text that can be retrieved so instead of having to aim with the mouse, I could just press something like NVDA+shift+T and hear "swap sides - tooltip" for that button.

Tooltips can often provide extra hints and explanations for things even when the item is correctly set up for NVDA to announce what it is, so just thought this would be useful for those who either do not use mice, or those of us who do not like having to aim precisely at small objects.

**Comment by josephsl** [comment]
Hi, sometimes, routing mouse position to the cursor position allows tool tips to show up, but that’s something I personally don’t use unless needed to demonstrate something or get clarifications (yes, I do use mouse and touch sometimes as a blind user). Thanks.

**Comment by leonardder** [comment]
@TechHorseG Shouldn't moving the mouse to navigator object command (NVDA+numpadDivide, laptop: NVDA+shift+m) be enough for your use case?

**Comment by TechHorseG** [comment]
Unfortunately it doesn't always work. In fact the buttons at the bottom of this edit box have tooltips but NVDA won't even read them out when I hover, and still won't when I press NVDA+shift+M. Some sort of direct retrieving of the tooltip text may help here.
    On Thursday, 17 October 2019, 14:31:08 BST, Leonard de Ruijter <notifications@github.com> wrote:  
 
 
@TechHorseG Shouldn't moving the mouse to navigator object command (NVDA+numpadDivide, laptop: NVDA+shift+m) be enough for your use case?

**Comment by Adriani90** [comment]
It seems this is not working with nvda+shift+m properly, a real mouse movement triggers the tooltip to appear. Is it possible to add a new option to NVDA mouse settings to automatically follow system focus? In that case the tooltip would be read automatically when the system focus lands on an element.
See #6227 for a concrete use case.

NVDA can fail the other way round to report tooltips, it reports them on system focus but not on mouse hover. See #3313 and #3314 for more details.

cc: @SaschaCowley, @jcsteh
